src/maker.rs

```diff
@@ -136,9 +124,14 @@ impl Maker {
             return Ok(TakeRequestDecision::CannotTradeWithTaker);
         }
 
+        if self.mid_market_rate.is_none() {
+            anyhow::bail!(RateNotAvailable(order.inner.position))
+        }
+
+        let current_mid_market_rate = self.mid_market_rate.unwrap();
```

Comment:
:wrench: If you use if you match or use `unwrap_or_else` above you can avoid this unwrap.